openBCIBoard.js

```diff
@@ -856,180 +1013,174 @@ function OpenBCIFactory() {
+                        if (this.options.verbose) console.log('Money!');
+                        if (this.options.verbose) console.log(data.toString());
                         this.isLookingForKeyInBuffer = false; // critical!!!
                         this.emit('ready'); // tell user they are ready to stream, etc...
+                    } else if (this.searchingBuf.equals(this.searchBuffers.timeSyncStart)) {
+                        this.sync.ntp1 = now();
+                        if(this.options.verbose) console.log('Got time sync request: ' + this.sync.npt1.toFixed(4));
+                        this.sync.ntp2 = now();
+                        this._writeAndDrain('<' + (this.sync.ntp1 * 1000) + (this.sync.ntp2 * 1000));
+                        this.searchingBuf = this.searchBuffers.miscStop;
+
                     } else {
                         getChannelSettingsObj(data.slice(i)).then((channelSettingsObject) => {
                             this.emit('query',channelSettingsObject);
                         }, (err) => {
                             console.log('Error: ' + err);
                         });
-                        this.searchingBuf = this.moneyBuf;
+                        this.searchingBuf = this.searchBuffers.miscStop;
                         break;
                     }
                 }
             }
         } else { // steaming operation should lead here...
-            // send input data to master buffer
-            this._bufMerger(data);
-
-            // parse the master buffer
-            while(this.masterBuffer.packetsRead < this.masterBuffer.packetsIn) {
-                var rawPacket = this._bufPacketStripper();
-                var newSample = openBCISample.convertPacketToSample(rawPacket);
-                if(newSample) {
-                    this.emit('rawDataPacket', rawPacket);
-                    newSample._count = this.sampleCount++;
-                    if(this.impedanceTest.active) {
-                        if (this.impedanceTest.onChannel != 0) {
-                            // Get an average of the impedance
-                            openBCISample.impedanceCalculationForChannel(newSample,this.impedanceTest.onChannel)
-                                .then(rawValue => {
-                                    impedanceTestApplyRaw.call(this,rawValue);
-                                }).catch(err => {
-                                    console.log('Impedance calculation error: ' + err);
+
+            // cbuffer
+            var bytesToRead = sizeOfData;
+
+            // is there old data?
+            if (this.buffer) {
+                // Get size of old buffer
+                var oldBufferSize = this.buffer.byteLength;
+
+                // Make a new buffer
+                var newDataBuffer = new Buffer(bytesToRead + oldBufferSize);
+
+                // Put old buffer in the front of the new buffer
+                var oldBytesWritten = this.buffer.copy(newDataBuffer);
```

Comment:
No, this line copies the contents of `this.buffer` into `newDataBuffer`. I think it's weird too how node makes the target buffer an input argument, counter intuitive for sure.